### PR TITLE
Split module into a non-sbt specific core, and an sbt plugin

### PR DESCRIPTION
**Pull request by Baccata**
Currently getting this error when running `sbt-plugin/scripted` : 

```
[info] [info] Cloning and doing the whole shebang
[error] scala.MatchError: List(A suitable version of cmake was not found (required v3.22.2). Downloading portable cmake v3.22.2...) (of class scala.collection.immutable.$colon$colon)
[error]         at com.indoorvivants.vcpkg.Vcpkg$Dependencies$.$anonfun$parse$1(Vcpkg.scala:88)
[error]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
```

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,11 +1,6 @@
-name := """sbt-vcpkg"""
-organization := "com.indoorvivants"
-
-sbtPlugin := true
-
 inThisBuild(
   List(
-    organization := "com.indoorvivants",
+    organization := "com.indoorvivants", // TODO : org should probably be com.indoorvivants.vcpkg
```
Comment (Baccata) [inline build.sbt:3]: Changing the org would make it easier to make sense of the stats you get from sonatype

File: build.sbt [hunk 2]
```diff
@@ -21,20 +16,47 @@ inThisBuild(
   )
 )
 
-initialCommands in console := """import com.indoorvivants.sbt._"""
+lazy val scala213 = "2.13.8"
+lazy val scala212 = "2.12.15"
+lazy val supportedScalaVersions = List(scala213, scala212)
```
Comment (Baccata) [inline build.sbt:21]: mill uses 2.13, therefore core needs to be cross-built

File: core/src/main/scala/com/indoorvivants/vcpkg/Vcpkg.scala [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
 import java.util.stream.Collectors
-import com.indoorvivants.vcpkg.sbt.Platform.OS.Linux
-import com.indoorvivants.vcpkg.sbt.Platform.OS.MacOS
-import com.indoorvivants.vcpkg.sbt.Platform.OS.Unknown
-import com.indoorvivants.vcpkg.sbt.Platform.OS.Windows
+import Platform.OS._
 
 class Vcpkg(binary: File, installation: File, debug: String => Unit = _ => ()) {
-  import sys.process.*
+  import sys.process._
```
Comment (Baccata) [inline core/src/main/scala/com/indoorvivants/vcpkg/Vcpkg.scala:11]: what's the scalac option that allows to use * as a wildcard in Scala 2.13 ?

File: core/src/main/scala/com/indoorvivants/vcpkg/package.scala [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.indoorvivants
+
+import java.io.File
+
+package object vcpkg {
+
+  private val FILE_SEPARATOR = System.getProperty("file.separator");
+
+  private[vcpkg] implicit class FileOps(private val file: File) extends AnyVal {
+    def /(s: String): File = new File(file.getAbsolutePath() + FILE_SEPARATOR + s)
```
Comment (Baccata) [inline core/src/main/scala/com/indoorvivants/vcpkg/package.scala:10]: not sure if that's the best way 🤷

File: core/src/main/scala/com/indoorvivants/vcpkg/Vcpkg.scala (hunk 3 above)
Comment (keynmol) [reply core/src/main/scala/com/indoorvivants/vcpkg/Vcpkg.scala:11]: `-source:3` I think?

Comment (keynmol) [reply core/src/main/scala/com/indoorvivants/vcpkg/Vcpkg.scala:11]: or `-Xsource:3`

File: core/src/main/scala/com/indoorvivants/vcpkg/package.scala (hunk 4 above)
Comment (keynmol) [reply core/src/main/scala/com/indoorvivants/vcpkg/package.scala:10]: I think you can do `Paths.get(file.toPath, s).toFile`

File: build.sbt (hunk 1 above)
Comment (keynmol) [reply build.sbt:3]: Yeah, good point

**Comment by Baccata** [comment]
Closing cause I'm a dumdum and targeted my fork

**Pull request closed without merging**